diff --git a/tests/ApiTokenFunctionalTest.php b/tests/ApiTokenFunctionalTest.php
index 217d3c9..29706e6 100644
--- a/tests/ApiTokenFunctionalTest.php
+++ b/tests/ApiTokenFunctionalTest.php
@@ -24,7 +24,7 @@ class ApiTokenFunctionalTest extends WebTestCase
 
         // X-TOKEN is the header name
         $client->request('GET', '/secure', [], [], [
-            'HTTP_X-TOKEN' => $apiToken->getToken().'foo',
+            'HTTP_X-TOKEN' => $apiToken->getToken(),
             'HTTP_ACCEPT' => 'application/json'
         ]);
 
